### DISCUSSION TITLE
Ability for admin to clean up /homes from deprecated worlds

### DISCUSSION THREAD

**Discussion by mrfloris** [Zrips/CMI#7492] [Enhancement]
Per Discord user `@JustinoBambino#0001`;

> not sure if it's possible now but deleting homes in certain worlds would be fantastic for like when resource worlds reset

Over time we might end up with a bunch of old worlds that we know for a fact will not return. To help clean up the database a bit for our players, it would be nice if we could purge some of the data such as homes from these worlds. 

I do really want homes to show if the world isn't there anymore. Because sometimes there are event worlds that return, or mine worlds get reset, etc etc. I don't wish that to change. 

But if there's like a "temp_Event_2019" world that is finally cleaned up and we will never return, the admin can use a cmi util command or cmi home with some param, to help clean up this data. The cmi resetdbfield is too aggressive for this of course.

`/cmi homes -clean:temp_Event_2019`
`/cmi util homes -delete:temp_Event_2019`

Something along those lines.

**Comment by MrCrazys** [comment]
> Per Discord user `@JustinoBambino#0001`;
> 
> > not sure if it's possible now but deleting homes in certain worlds would be fantastic for like when resource worlds reset
> 
> Over time we might end up with a bunch of old worlds that we know for a fact will not return. To help clean up the database a bit for our players, it would be nice if we could purge some of the data such as homes from these worlds.
> 
> I do really want homes to show if the world isn't there anymore. Because sometimes there are event worlds that return, or mine worlds get reset, etc etc. I don't wish that to change.
> 
> But if there's like a "temp_Event_2019" world that is finally cleaned up and we will never return, the admin can use a cmi util command or cmi home with some param, to help clean up this data. The cmi resetdbfield is too aggressive for this of course.
> 
> `/cmi homes -clean:temp_Event_2019` `/cmi util homes -delete:temp_Event_2019`
> 
> Something along those lines.

Not a bad idea. Maybe it could be added as a parameter to `/cmi resetdbfields`, perhaps as something like `-w:[worldName],[worldName]`.

**Comment by TomLewis** [comment]
If they use SQL they could just write a query to do this.
